### PR TITLE
[Page] Add minimum height on page header

### LINKED ISSUE #4773: [Page] Add minimum height on page header

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Fixes #[4773](https://github.com/Shopify/polaris-react/issues/4773).

By not setting a minimum height, the page header height would visibly jump depending on if the `Page` component had actions and/or breadcrumbs or not. 

### WHAT is this pull request doing?

Sets a minimum height keeps the page header spacing consistent.
I didn't use `rem(36px)` in light of changes in the `v8.0.0-major` branch that removes that function.

Additionally, the height remains unaltered on mobile view (meaning page header jumps depending on content). I had an offline discussion with @sarahill about solutions for keeping it consistent but we agreed that would be out of scope and hot-fixes we tried didn't align with our density improvement initiative. 
    <details>
      <summary>Before gif — Inconsistent spacing on page header depending on content</summary>
      Before gif showing inconsistent spacing on page header depending on content
    </details>
    <details>
      <summary>After gif — Consistent spacing on page header regardless of content</summary>
      Gif showing consistent spacing on page header regardless of content
    </details>

## 

### How to 🎩

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code> :</summary>

```jsx
import React from 'react';

import {Page, Badge, Card} from '../src';

export function Playground() {
  return (
    // Page with actions and breadcrumbs -- uncomment below to test
    // <Page
    //   breadcrumbs={[{content: 'Products', url: '/products'}]}
    //   title="Jar With Lock-Lid"
    //   titleMetadata={<Badge status="attention">Verified</Badge>}
    //   primaryAction={{content: 'Save', disabled: true}}
    //   secondaryActions={[
    //     {content: 'Duplicate'},
    //     {content: 'View on your store'},
    //   ]}
    //   pagination={{
    //     hasPrevious: true,
    //     hasNext: true,
    //   }}
    // >
    //   <Card title="Page content" />
    // </Page>

    // Page with no actions and breadcrumbs -- comment out below if testing above component
    <Page title="Page with no actions/breadcrumbs">
      <Card title="Page content" />
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [x] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -6,9 +6,16 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
 
 ### Enhancements
 
-- Removed animtion from `Skeleton` components ([#4697](https://github.com/Shopify/polaris-react/pull/4697))
```
Comment (laurkim) [inline UNRELEASED.md:9]: removed because it was duplicated

File: UNRELEASED.md [hunk 2]
```diff
@@ -6,7 +6,14 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
...
```
Comment (laurkim) [inline UNRELEASED.md]: deleted because it was duplicated in latest update to `main`

**Review by sarahill — APPROVED** [review]
❇️ 🚢

File: UNRELEASED.md [hunk 3]
```diff
@@ -6,8 +6,9 @@ Use [the changelog guidelines](/documentation/Versioning%20and%20changelog.md) t
...
-- Remove duplicate duration(fast) usage. ([#4682](https://github.com/Shopify/polaris-react/pull/4682))
```
Comment (laurkim) [inline UNRELEASED.md]: Updated to make it past tense

File: UNRELEASED.md (hunk 2 above)
Comment (laurkim) [reply UNRELEASED.md]: correction — this was updated to correct a typo

**Review by chloerice — APPROVED** [review]
🚢 🚢 🚢

**Comment by ghost** [comment]
🎉 Thanks for your contribution to Polaris React!

**Pull request merged**